sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1187,7 +1231,13 @@ private long doForceInsertExistingDocumentWithHistory(SQLDatabase db,BasicDocume
         Preconditions.checkArgument(revisions.size() > 0, "Revision history should have at least one revision." );
 
         // First look up all locally-known revisions of this document:
-        DocumentRevisionTree localRevs = getAllRevisionsOfDocumentInQueue(db, newRevision.getId());
+
+        DocumentRevisionTree localRevs;
+        try {
+            localRevs = getAllRevisionsOfDocumentInQueue(db, newRevision.getId());
+        } catch (DocumentNotFoundException e){
+            throw new RuntimeException("This should not be thrown, library programming error", e);
```

Comment:
Explain in a comment why it shouldn't be thrown, and possibly make a better log message so people can file a bug report if they encounter it.